futures/examples/async_context.rs

```diff
@@ -0,0 +1,62 @@
+#![feature(async_await, futures_api, await_macro, pin, arbitrary_self_types)]
+extern crate futures;
+extern crate rand;
+
+use futures::{
+    executor //standard executors to provide a context for futures and streams
+};
+
+fn main() {
+    //
+    // async implements future, so will be passed a std::task::Context when
+    // future::poll is invoked. Once the context is passed, it will then be
+    // available within the async block for additional async/await operations.
+    //
+    // This is approximately equal to having a thread local context that is
+    // available within the future, and given to additional async/await blocks.
+    //
+    // ```text
+    //   static mut TASK_CONTEXT: Option<NonNull<std::task::Context>> = None;
+    //   let polled = unsafe {
+    //     let mut cx = TASK_CONTEXT.take().unwrap();
+    // ```
```

Comment:
- `async` is just the keyword. You mean "async block"
- The example with the thread local context is well intentioned, but I think it will unfortunately confuse newbies more than help them